src/nvim/api/fold.c

```diff
@@ -0,0 +1,43 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "nvim/api/fold.h"
+#include "nvim/api/private/dispatch.h"
+#include "nvim/api/private/validate.h"
+#include "nvim/fold.h"
+#include "nvim/pos_defs.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "api/fold.c.generated.h"  // IWYU pragma: keep
+#endif
+
+/// Gets the start and end lines for the closed fold on the given line
+///
+/// @param window   |window-ID|, or 0 for current window
+/// @param lnum   The line number to check (0-indexed)
+/// @param[out] err Error details, if any
+/// @return  Dict containing fold information, with these keys:
+///          - first: The start line of the current closed fold, or `nil` if none
+///          - last: The end line of the current closed fold, or `nil` if none
+Dict(fold_info) nvim__fold_info(Window window, Integer lnum, Arena *arena, Error *err)
```

Comment:
sorry 😅 , one more... if/when we graduate this, presumably it will always target a window, so should name it `nvim_win_get_fold`. 

(Though "buffer local" folds could make sense in theory, then should name this `nvim_fold_get({ win?: int, buf?: int})`. But that's probably unlikely).

```suggestion
Dict(fold_info) nvim__win_get_fold(Window window, Integer lnum, Arena *arena, Error *err)
```

Reply:
That makes sense; should this be moved to `api/window.c`? or just kept here

Reply:
fold.c probably is good to have, since we likely want a lot more fold api stuff